pkg/controller/velerobackup/hiveobject_test.go

```diff
@@ -0,0 +1,176 @@
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func TestNewHiveObject(t *testing.T) {
+	tests := []struct {
+		name               string
+		object             runtime.Object
+		logger             log.FieldLogger
+		expectedHiveObject *hiveObject
+		expectedError      error
+	}{
+		{
+			name:   "Only Valid ClusterDeployment",
+			object: testclusterdeployment.Build(unchangedClusterDeploymentBase()),
+			expectedHiveObject: &hiveObject{
+				object:   testclusterdeployment.Build(unchangedClusterDeploymentBase()),
+				checksum: testclusterdeployment.Build(unchangedClusterDeploymentBase()).Annotations[controllerutils.LastBackupAnnotation],
+			},
+		},
+		{
+			name:   "Only Valid SyncSet",
+			object: testsyncset.Build(unchangedSyncSetBase()),
+			expectedHiveObject: &hiveObject{
+				object:   testsyncset.Build(unchangedSyncSetBase()),
+				checksum: testsyncset.Build(unchangedSyncSetBase()).Annotations[controllerutils.LastBackupAnnotation],
+			},
+		},
+		{
+			name:               "Invalid type (not a hive object)",
+			object:             &corev1.Pod{},
+			expectedHiveObject: nil,
+			expectedError:      fmt.Errorf("Unknown Type: *v1.Pod"),
+		},
+		{
+			name:               "Invalid nil object",
+			object:             nil,
+			expectedHiveObject: nil,
+			expectedError:      fmt.Errorf("Unknown Type: <nil>"),
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			// Arrange
+
+			// Act
+			actualHiveObject, actualError := newHiveObject(test.object, test.logger)
+
+			// Assert
+			assert.Equal(t, test.expectedHiveObject, actualHiveObject)
+			assert.Equal(t, test.expectedError, actualError)
+		})
+	}
+}
+
+func TestHasChanged(t *testing.T) {
+	tests := []struct {
+		name            string
+		hiveObject      *hiveObject
+		expectedChanged bool
+		checksumFunc    controllerutils.ChecksumOfObjectsFunc
+	}{
+		{
+			name:            "Unchanged ClusterDeployment",
+			hiveObject:      ro2ho(testclusterdeployment.Build(unchangedClusterDeploymentBase())),
+			expectedChanged: false,
+			checksumFunc:    controllerutils.GetChecksumOfObjects,
+		},
+		{
+			name:            "Changed ClusterDeployment",
+			hiveObject:      ro2ho(testclusterdeployment.Build(changedClusterDeploymentBase())),
+			expectedChanged: true,
+			checksumFunc:    controllerutils.GetChecksumOfObjects,
+		},
+		{
+			name:            "Unchanged SyncSet",
+			hiveObject:      ro2ho(testsyncset.Build(unchangedSyncSetBase())),
+			expectedChanged: false,
+			checksumFunc:    controllerutils.GetChecksumOfObjects,
+		},
+		{
+			name:            "Changed SyncSet",
+			hiveObject:      ro2ho(testsyncset.Build(changedSyncSetBase())),
+			expectedChanged: true,
+			checksumFunc:    controllerutils.GetChecksumOfObjects,
+		},
+		{
+			name:            "Error from Checksum func",
```

Comment:
This subtest is not testing what the name implies it should be testing. This test is redundant with the "Unchanged SyncSet" subtest.